<?xml version="1.0" encoding="utf-8"?>
<!--launch file for smoke package.-->
<launch>

  <!-- ROS and network parameter files -->
  <arg name="ros_param_file" default="$(find smoke)/config/darknet_svm_rbx1.yml"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="smoke" file="$(arg ros_param_file)"/>

  <!--Launching camera nodes-->
  <include file="$(find smoke)/launch/usb_cam.1.launch"></include>

  <!--robot bringup and navigation-->
  <include file="$(find rbx1_bringup)/launch/fake_turtlebot.launch"></include>
  <include file="$(find rbx1_nav)/launch/fake_amcl.launch"></include>
  <!--include file="$(find turtlebot_bringup)/launch/minimal.launch"></include>
  <include file="$(find turtlebot_navigation)/launch/amcl_demo.launch"></include-->

  <!--launching darknet_ros-->
  <include file="$(find smoke)/launch/darknet_ros_smoke.launch"></include>

  <!--neural network classifier-->
  <node pkg="smoke" type="smoke_nn_server.py" name="smoke_nn_server_node" output="screen"/>

  <!--svm classifier-->
  <node pkg="smoke" type="smoke_svm_server" name="smoke_svm_server_node" output="screen"/>

  <!--darknet_detetor-->
  <node pkg="smoke" type="darknet_detector" name="darknet_detector_node" output="screen" required="true"/>

  <!--alarm-->
  <node pkg="smoke" type="alarm_sub.py" name="alarm_sub_node" output="log" />
  <!--node pkg="smoke" type="weixin_alarm.py" name="weixin_alarm_node" output="log" /-->

  <!--robot action-->
  <node pkg="smoke" type="follow_the_route.py" name="cycle_around_node" output="log" />
  <node pkg="smoke" type="kobuki_actSrv.py" name="kobuki_action_node" output="log" />

</launch>
